authors/apps/profiles/views.py

```diff
@@ -41,13 +43,86 @@ def put(self, request, *args, **kwargs):
             serializer = self.serializer_class(profile,
-                                           data=request.data['profile'])
+                                               data=request.data['profile'],
+                                               context={'request': request})
             serializer.is_valid(raise_exception=True)
             serializer.save(data=request.data)
-            return Response(UpdateProfileSerializer(profile).data)
+            return Response(serializer.data)
         return Response(
             data={'message':
                     'You have no permissions to edit others profile.'},
             status=status.HTTP_403_FORBIDDEN
         )
 
+def get_user_object(username):
+    # get user object
+    try:
+        return User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise NotFound(detail='User not found.')
+
+
+class FollowView(APIView):
+    #Function to handle follow user functionality
+    permission_classes = (IsAuthenticated,)
+
+    def get_follow_object(self, follower, followed):
+        # Method to get a follow object
+        try:
+            return Follow.objects.get(follower=follower, followed=followed)
+        except Follow.DoesNotExist:
+            return None
+
+    def post(self, request, username):
+        # post to follow a user
+        user = get_user_object(username)
+        if user == request.user:
+            return Response({'error': 'Sorry, you can not follow your self.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        if not self.get_follow_object(request.user, user):
+            Follow(followed=user, follower=request.user).save()
+        message = f"You are following {user.username}."
+        return Response({'message': message}, status=status.HTTP_200_OK,)
+
+    def delete(self, request, username):
+        # unfollow a user
+        user = get_user_object(username)
+        follow_Obj = self.get_follow_object(request.user, user)
+        if not follow_Obj:
+            return Response({'error': f'You are not following {user.username}.'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        follow_Obj.delete()
+        message = f"You have successfully unfollowed {user.username}."
+        return Response({'message': message}, status=status.HTTP_200_OK,)
+
+
+class Followers(APIView):
+    """ Return users that follow the provided username"""
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, username):
+        # get the user's followers
+        user = get_user_object(username)
+        followers = Follow.objects.filter(followed=user)
+        profiles = []
+        for value in followers:
+            profile = UserProfile.objects.get(user=value.follower)
+            profile = FetchUserProfileSerializer(profile)
+            profiles.append(profile.data)
+        return Response(profiles, status=status.HTTP_200_OK,)
```

Comment:
@joelmugaya We should return a message `You have no followers` when the list being returned empty.